### PR TITLE
Support for SQL Server sparse columns

### LINKED ISSUE #8023: Support for SQL Server sparse columns
I know I can do this with raw SQL in Up method, but can this be done with fluent API.
If it can How, and if not, are there plans to add this feature in future versions?

### Further technical details
EF Core version: 1.1.1
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: Visual Studio 2017

### PR DESCRIPTION
**Pull request by roji**
Sometimes you just got to do a "good first issue" on a Saturday morning.

This should be useful with TPH mapping.

Closes #8023

### CODE REVIEW

File: src/EFCore.SqlServer/Migrations/SqlServerMigrationsSqlGenerator.cs [hunk 1]
```diff
@@ -1463,13 +1463,34 @@ protected override void Generate(UpdateDataOperation operation, IModel model, Mi
...
             Check.NotNull(builder, nameof(builder));
 
-            base.ColumnDefinition(
-                schema,
-                table,
-                name,
-                operation,
-                model,
-                builder);
+            if (operation.ComputedColumnSql != null)
```
Comment (roji) [inline src/EFCore.SqlServer/Migrations/SqlServerMigrationsSqlGenerator.cs:1466]: Unfortunately SPARSE must appear after the type but before NULL/NOTNULL :cry:

**Comment by ErikEJ** [comment]
That is a pretty extensive "first issue" :smile:

**Comment by roji** [comment]
@ErikEJ you're right... Adding a simple bool "migrations annotation" all the way isn't actually that simple... Maybe we can point to this as an example for other similar features.

File: src/EFCore.SqlServer/Extensions/SqlServerPropertyExtensions.cs [hunk 2]
```diff
@@ -489,5 +489,49 @@ public static bool IsCompatibleWithValueGeneration([NotNull] IProperty property)
...
+        /// <summary>
+        ///     Returns a value indicating whether the property's column is sparse.
+        /// </summary>
+        /// <param name="property"> The property. </param>
+        /// <returns> <see langword="true" /> if the property's column is sparse. </returns>
+        public static bool? IsSparse([NotNull] this IProperty property)
+            => (bool?)property[SqlServerAnnotationNames.Sparse];
+
+        /// <summary>
+        ///     Sets a value indicating whether the property's columns is sparse.
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Extensions/SqlServerPropertyExtensions.cs:502]:
```suggestion
        ///     Sets a value indicating whether the property's column is sparse.
```

**Review by AndriySvyryd — APPROVED** [review]

**Pull request merged**